src/main/kotlin/com/hypto/iam/server/db/repositories/UserRepo.kt

```diff
@@ -69,12 +68,10 @@ object UserRepo : BaseRepo<UsersRecord, Users, String>() {
         .and(USERS.VERIFIED.eq(true))
         .fetchOne()
 
-    suspend fun findByAliasUsername(aliasUsername: String): UsersRecord? {
-        val conditions = DSL.or(aliasUsernames.map { it.eq(aliasUsername) })
-
+    suspend fun findByPreferredUsername(preferredUsername: String): UsersRecord? {
         return ctx("users.findByAliasUsername")
             .selectFrom(USERS)
-            .where(conditions)
+            .where(USERS.PREFERRED_USERNAME.eq(preferredUsername))
```

Comment:
Using "union all" is also one of the suggested approach to query with or clause.

https://dba.stackexchange.com/questions/172907/multicolumn-indexes-and-or-clause

Reply:
This way need not even query the preferred_username column. UNION can cost more as it checks both columns, merges them and then applies conditions like deleted=false, verified=true.